jujugui/static/gui/src/app/app.js

```diff
@@ -1240,13 +1240,14 @@ YUI.add('juju-gui', function(Y) {
     */
     _generateSocketUrl: function(callback) {
       this.jem = null;
-      if (this.get('jemUrl')) {
+      if (this.get('jemURL')) {
         var bakery = new Y.juju.environments.web.Bakery({
           webhandler: new Y.juju.environments.web.WebHandler(),
           interactive: this.get('interactiveLogin'),
           serviceName: 'jem'
         });
-        this.jem = new window.jujulib.environment(this.get('jemUrl'), bakery);
+        this.jem = new window.jujulib.environment(
+            this.get('jemURL'), this.get('jemAPIPath'), bakery);
```

Comment:
Oh, good catch.

I note also that charmstore's parameters have always come from juju_config--we don't really want to do that, right?